Expensify/App
# [No QA] Add context and more specific logs for various fetch errors

### Details
`fetch()` throws all sorts of errors and in an effort to understand what they might mean I have been looking into them and adding more context about them, whether they can be retried, etc. This is in part related to the changes in Network Improvements.

- Improve logging
- Adds the context of the `reponse.status` which is missing
- Throws some errors that we temporarily stopped throwing to learn more about them

**To do:**
- [x] Catch and throw Auth timeouts
- [x] Catch and throw Auth down errors

### Fixed Issues
$ https://github.com/Expensify/App/issues/8517

### Tests
1. Sign in on dev
2. Kill Auth `sudo service auth stop`
3. Verify Auth is down errors are getting handled and requests are retried for a bit before things stop
4. Start Auth up again
5. Simulate a bedrock timeout error by throwing `555 Timeout` somewhere in the auth code (like `createAction()`)
```diff
diff --git a/auth/lib/Report.cpp b/auth/lib/Report.cpp
index a22b0ee95..3fc3870c1 100644
--- a/auth/lib/Report.cpp
+++ b/auth/lib/Report.cpp
@@ -194,6 +194,8 @@ void Report::addAutoAction(SQLite& db, const int64_t reportID, const int64_t acc

 string Report::createAction(SQLite& db, const int64_t reportID, const int64_t requestingAccountID, const int64_t actionAccountID, const string& action, const string& message, const int64_t status, const string& authToken)
 {
+    STHROW("555 Timeout");
+
```
6. Build auth and try to leave a comment on a report
7. Verify that the timeout error is caught as a `Socket` error and retried


### PR Review Checklist

<details>
<summary><h4>PR Reviewer Checklist</h4></summary>


</details>

### QA Steps

No QA
